staging/src/k8s.io/pod-security-admission/test/run.go

```diff
@@ -82,22 +83,64 @@ func checksForLevelAndVersion(checks []policy.Check, level api.Level, version ap
 	return retval, nil
 }
 
-// maxMinorVersionToTest returns the maximum minor version to exercise for a given set of checks.
-// checks are assumed to be well-formed and valid to pass to policy.NewEvaluator().
-func maxMinorVersionToTest(checks []policy.Check) (int, error) {
-	// start with the release under development (1.22 at time of writing).
-	// this can be incremented to the current version whenever is convenient.
-	maxTestMinor := 22
+// computeVersionsToTest returns all the versions that have distinct checks defined,
+// all the versions that have distinct minimal valid pod fixtures defined, and
+// any hard-coded versions that should always be tested.
+//
+// This lets us sparsely test all versions with distinct fixture or policy changes
+// without needing to exercise every intermediate version that had no changes.
+func computeVersionsToTest(t *testing.T, checks []policy.Check) []api.Version {
+	seenVersions := map[api.Version]bool{}
+	versions := []api.Version{}
+
+	// include all versions we have registered distinct checks for
 	for _, check := range checks {
-		lastCheckVersion := check.Versions[len(check.Versions)-1].MinimumVersion
-		if lastCheckVersion.Major() != 1 {
-			return 0, fmt.Errorf("expected major version 1, got %d", lastCheckVersion.Major())
+		for _, checkVersion := range check.Versions {
+			if checkVersion.MinimumVersion.Major() != 1 {
+				t.Fatalf("expected major version 1, got %d", checkVersion.MinimumVersion.Major())
+			}
+			if seenVersions[checkVersion.MinimumVersion] {
+				continue
+			}
+			seenVersions[checkVersion.MinimumVersion] = true
+			versions = append(versions, checkVersion.MinimumVersion)
+		}
+	}
+	if len(versions) == 0 {
+		t.Fatal("no versions defined for checks")
+	}
+
+	// include all versions we have registered distinct fixtures for
+	for _, versionsForLevel := range minimalValidPods {
+		for version := range versionsForLevel {
+			if version.Major() != 1 {
+				t.Fatalf("expected major version 1, got %d", version.Major())
+			}
+			if seenVersions[version] {
+				continue
+			}
+			seenVersions[version] = true
+			versions = append(versions, version)
 		}
-		if lastCheckVersion.Minor() > maxTestMinor {
-			maxTestMinor = lastCheckVersion.Minor()
+	}
+
+	alwaysIncludeVersions := []api.Version{
+		// include the oldest version by default
+		api.MajorMinorVersion(1, 0),
+		// include the release under development (1.22 at time of writing).
+		// this can be incremented to the current version whenever is convenient.
+		api.MajorMinorVersion(1, 22),
+	}
+	for _, version := range alwaysIncludeVersions {
+		if seenVersions[version] {
+			continue
 		}
+		seenVersions[version] = true
+		versions = append(versions, version)
```

Comment:
nit: I'd prefer to just construct `versions` at the end by iterating over `seenVersions`

Reply:
done